Modules/services/Model/SeServiceType.php

```diff
@@ -9,6 +9,61 @@
  */
 class SeServiceType extends Model {
 
+    private static $serviceTypes;
+
+     /**
+     * Constructor
+     */
+    public function __construct() {
+        self::$serviceTypes[1] = "Quantity";
+        self::$serviceTypes[2] = "Time minutes";
+        self::$serviceTypes[3] = "Time hours";
+        self::$serviceTypes[4] = "Price";
+        self::$serviceTypes[5] = "Half day";
+        self::$serviceTypes[6] = "Day";
+    }
+
+    public function updateServiceTypesReferences() {
+        // extract service types names and ids from database
+        $sql = "SELECT `id`, `name` FROM se_service_types";
+        $typesData = $this->runRequest($sql, array())->fetchAll();
+        $tmpTypes = [];
+        $modifiedServiceIds = [];
+        $serviceTypesArray = self::$serviceTypes;
+
+        for ($i=0; $i<6; $i++) {
+            // format data get from database to match $serviceTypes format for further comparisons
+            $tmpTypes[$typesData[$i]["id"]] = $typesData[$i]["name"];
```

Comment:
Name or local_name?

Reply:
Name. I replaced local_name values by a translation function in servicesTranslator.php. More efficient and, for an upgrade, it allows to easily find out what is the type, whether it was set as local_name or name in db.

Reply:
So here, once $tmpTypes filled, I can compare its values (names from db) with (values from static array $serviceTypes or translated values from static array $serviceTypes). So we get the right corresponding indexes. Then translation is used, as usual, to adapt displaying to the user's language settings